WooCommerce/Classes/Authentication/Store Creation/Profiler/RequiredStoreCreationProfilerQuestionView.swift

```diff
@@ -0,0 +1,90 @@
+import Combine
+import SwiftUI
+
+/// Handles the navigation action and provides the continue button state in a required profiler question view during store creation.
+/// The question is not skippable.
+protocol RequiredStoreCreationProfilerQuestionViewModel {
+    /// Called when the continue button is tapped.
+    func continueButtonTapped() async
+
+    /// Whether the continue button is enabled for the user to continue.
+    var isContinueButtonEnabled: AnyPublisher<Bool, Never> { get }
+}
+
+/// Shows a mandatory profiler question during the store creation flow with a generic content.
+struct RequiredStoreCreationProfilerQuestionView<QuestionContent: View>: View {
+    private let viewModel: StoreCreationProfilerQuestionViewModel & RequiredStoreCreationProfilerQuestionViewModel
+    @ViewBuilder private let questionContent: () -> QuestionContent
+    @State private var isWaitingForCompletion: Bool = false
+    @State private var isContinueButtonEnabled: Bool = false
+
+    init(viewModel: StoreCreationProfilerQuestionViewModel & RequiredStoreCreationProfilerQuestionViewModel,
+         @ViewBuilder questionContent: @escaping () -> QuestionContent) {
+        self.viewModel = viewModel
+        self.questionContent = questionContent
+    }
+
+    var body: some View {
+        ScrollView {
+            StoreCreationProfilerQuestionView<QuestionContent>(viewModel: viewModel, questionContent: questionContent)
+        }
+        .safeAreaInset(edge: .bottom) {
+            VStack {
+                Divider()
+                    .frame(height: Layout.dividerHeight)
+                    .foregroundColor(Color(.separator))
+                Button(Localization.continueButtonTitle) {
+                    Task { @MainActor in
+                        isWaitingForCompletion = true
```

Comment:
❓ Is there a reason we can't move the `isWaitingForCompletion` property to the view model?

Reply:
similar to the answer to the other question https://github.com/woocommerce/woocommerce-ios/pull/8509#discussion_r1060286040, if moving `isWaitingForCompletion` to the view model protocol then it has to be bound to another state variable. also, this property is specific to the UI and doesn't require extra logic from the view model so I thought it makes sense to stay in the view.